components/cldr-json-data-provider/src/transform/dates.rs

```diff
@@ -0,0 +1,399 @@
+                langid: l.langid.clone(),
+            })
+            .collect();
+        Ok(Box::new(list.into_iter()))
+    }
+}
+
+impl From<&cldr_json::StylePatterns> for gregory::patterns::StylePatternsV1 {
+    fn from(other: &cldr_json::StylePatterns) -> Self {
+        Self {
+            full: other.full.clone(),
+            long: other.long.clone(),
+            medium: other.medium.clone(),
+            short: other.short.clone(),
+        }
+    }
+}
+
+impl From<&cldr_json::Dates> for gregory::DatesV1 {
+    fn from(other: &cldr_json::Dates) -> Self {
+        Self {
+            symbols: gregory::DateSymbolsV1 {
+                months: (&other.calendars.gregorian.months).into(),
+                weekdays: (&other.calendars.gregorian.days).into(),
+                day_periods: (&other.calendars.gregorian.day_periods).into(),
+            },
+            patterns: gregory::PatternsV1 {
+                date: (&other.calendars.gregorian.date_formats).into(),
+                time: (&other.calendars.gregorian.time_formats).into(),
+                date_time: (&other.calendars.gregorian.date_time_formats).into(),
+            },
+        }
+    }
+}
+
+macro_rules! symbols_from {
+    ([$name: ident, $name2: ident], [ $($element: ident),* ]) => {
+        impl From<&cldr_json::$name::Symbols> for gregory::$name2::SymbolsV1 {
+            fn from(other: &cldr_json::$name::Symbols) -> Self {
+                Self([
+                    $(
+                        other.$element.clone(),
+                    )*
+                ])
+            }
+        }
+        symbols_from!([$name, $name2]);
+    };
+    ([$name: ident, $name2: ident], { $($element: ident),* }) => {
+        impl From<&cldr_json::$name::Symbols> for gregory::$name::SymbolsV1 {
+            fn from(other: &cldr_json::$name::Symbols) -> Self {
+                Self {
+                    $(
+                        $element: other.$element.clone(),
+                    )*
+                }
+            }
+        }
+        symbols_from!([$name, $name]);
+    };
+    ([$name: ident, $name2: ident]) => {
+        impl From<&cldr_json::$name::Contexts> for gregory::$name2::ContextsV1 {
+            fn from(other: &cldr_json::$name::Contexts) -> Self {
+                Self {
+                    format: (&other.format).into(),
+                    stand_alone: other.stand_alone.as_ref().and_then(|stand_alone| {
+                        if other.format == *stand_alone {
+                            None
+                        } else {
+                            Some(stand_alone.into())
+                        }
+                    })
+                }
+            }
+        }
+
+        impl PartialEq<cldr_json::$name::StandAloneWidths> for cldr_json::$name::FormatWidths {
+            fn eq(&self, other: &cldr_json::$name::StandAloneWidths) -> bool {
+                other.abbreviated.is_none() || Some(&self.abbreviated) == other.abbreviated.as_ref() &&
```

Comment:
I think you need some more `()` due to order of operations.

Can you add a test case for this behavior?

Reply:
Thanks! I remodelled it to bring a bit more unaliasing via ability to skip widths based on if they differ in `stand_alone`, and added a unit test.